src/components/Table/Table/__stories__/table.stories.mdx

```diff
@@ -0,0 +1,187 @@
+
+export const metaSettings = createStoryMetaSettingsDecorator({
+  component: Table,
+})
+
+<Meta
+  title="Data display/Table"
+  component={ Table }
+  argTypes={ metaSettings.argTypes }
+  decorators={ metaSettings.decorators }
+/>
+
+<!--- Component template -->
+
+export const tableTemplate = createComponentTemplate(Table);
+
+<!--- Component documentation -->
+
+# Table
+- [Overview](#overview)
+- [Props](#props)
+- [Loading](#loading)
+- [Empty State](#empty-state)
+- [Scroll](#scroll)
+
+## Overview
+Table
+<Canvas>
+  <Story name="Overview">
+    {() => {
+      const [tableData, setTableData] = useState(emailTableData);
+      const [sorting, setSorting] = useState({});
+      const onSort = (columnId, sortState) => {
+        setSorting({ [columnId]: sortState });
+        setTableData(sort(columnId, sortState, tableData));
+      };
+      return (
+        <Table
+          dataState={{ isError: false, hasMoreResults: false }}
+          errorState={<TableErrorState />}
+          emptyState={<TableEmptyState />}
+          columns={emailColumns}
+        >
+          <TableHeader>
+            {emailColumns.map((headerCell, index) => (
+              <TableHeaderCell
+                key={index}
+                title={headerCell.title}
+                onSortClicked={sortState => onSort(headerCell.id, sortState)}
+                sortState={sorting[headerCell.id]}
+              ></TableHeaderCell>
+            ))}
+          </TableHeader>
+          <TableBody>
+            {tableData.map(rowItem => (
+              <TableRow key={rowItem.id}>
+                <TableCell>{rowItem.sentOn}</TableCell>
+                <TableCell>{rowItem.subject}</TableCell>
+                <TableCell>
+                  <TableAvatar text={rowItem.sentBy} />
+                </TableCell>
+                <TableCell>
+                  <Label text={rowItem.status} isAnimationDisabled />
+                </TableCell>
+                <TableCell>{rowItem.emailsSent}</TableCell>
+              </TableRow>
+            ))}
+          </TableBody>
+        </Table>
+      );
+    }};
+  </Story>
+</Canvas>
+
+## Props
+<ArgsTable of={ Table } />
+
+## Loading
+Using skeleton
+<Canvas>
+  <Story name="Loading">
+    {() => {
+      return (
+        <Table
+          dataState={{ isError: false, hasMoreResults: false }}
+          errorState={<TableErrorState />}
+          emptyState={<TableEmptyState />}
+          columns={emailColumns}
+        >
+          <TableHeader>
+            {emailColumns.map((headerCell, index) => (
+              <TableHeaderCell
+                key={index}
+                title={headerCell.title}
+              ></TableHeaderCell>
+            ))}
+          </TableHeader>
+          <TableBody>
+            {[...new Array(5)].map((key, index) => (
+              <TableRow key={key}>
+                <TableCell>
+                  <Skeleton type={Skeleton.types.TEXT} height={20} width={index % 2 === 0 ? 140 : 70} />
```

Comment:
I remember that we talked about this before, but now I wondering if there is a way to reduce boilerplate here... let's talk about it in our meeting